aten/src/ATen/native/native_functions.yaml

```diff
@@ -1829,6 +1830,7 @@
 - func: smm(Tensor self, Tensor mat2) -> Tensor
   variants: function, method
 
+# softmax allows positional dtype, unlike most operators, because kwonly is BC-breaking when loading jit models.
```

Comment:
For my understanding: why is kw-only BC-breaking when loading JIT models? Does the JIT trace positional-kw args as positional args?

Reply:
Yup. The issue description that caused us to revert (also sent you a link internally):

`F.log_softmax` was tracing to either `torch.log_softmax(logits, dim)` if dtype was not provided, or `torch.log_softmax(logits, dim, dtype)` if dtype wasn't provided.
It's now tracing to either `torch.log_softmax(logits, dim, dtype=annotate(Optional[int], None))` or `torch.log_softmax(logits, dim, dtype=dtype)` respectively.

Reply:
Cool. Might be good to put that into the comment so that if someone else tries to do the change in the future they won't have to look up this pull request.

Reply:
yeah I kinda want to avoid being excessively verbose though. I think if we add a unit test (https://github.com/pytorch/pytorch/issues/22650), that'll add clearer protection.

Reply:
Yeah, unit test is great. In general TorchScript should probably have BC unit tests like this